pkg/apis/networking/validation/validation.go

```diff
@@ -487,3 +503,10 @@ func allPathsPassRegexValidation(ingress *networking.Ingress) bool {
 	}
 	return true
 }
+
+func resourceBackendPresent(ingress *networking.Ingress) bool {
+	if ingress.Spec.Backend != nil && ingress.Spec.Backend.Resource != nil {
+		return true
+	}
+	return false
```

Comment:
also need to iterate over paths to see if any of them point at a resource backend (like the iteration in allPathsPassRegexValidation)

add a unit test of resourceBackendPresent covering these scenarios:
* nil spec.Backend and no paths -> false
* nil spec.Backend.Resource and no paths -> false
* non-nil spec.Backend.Resource and no paths -> true
* nil spec.Backend, one rule with nil HTTP -> false
* nil spec.Backend, one rule with non-nil HTTP, no paths -> false
* nil spec.Backend, one rule with non-nil HTTP, one path with nil Backend.Resource -> false
* nil spec.Backend, one rule with non-nil HTTP, one path with non-nil Backend.Resource -> true